x/iscn/types/iscnmsg_test.go

```diff
@@ -0,0 +1,154 @@
+package types
+
+import (
+	"testing"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// priv1 = secp256k1.GenPrivKey()
+	// addr1 = sdk.AccAddress(priv1.PubKey().Address())
+	addr1 = "cosmos1y54exmx84cqtasvjnskf9f63djuuj68p7hqf47"
+
+	fingerprint1 = "hash://sha256/9564b85669d5e96ac969dd0161b8475bbced9e5999c6ec598da718a3045d6f2e"
+
+	stakeholder1 = IscnInput(`
+{
+	"entity": {
+		"@id": "did:cosmos:5sy29r37gfxvxz21rh4r0ktpuc46pzjrmz29g45",
+		"name": "Chung Wu"
+	},
+	"rewardProportion": 95,
+	"contributionType": "http://schema.org/author"
+}`)
+
+	stakeholder2 = IscnInput(`
+{
+	"rewardProportion": 5,
+	"contributionType": "http://schema.org/citation",
+	"footprint": "https://en.wikipedia.org/wiki/Fibonacci_number",
+	"description": "The blog post referred the matrix form of computing Fibonacci numbers."
+}`)
+
+	contentMetadata1 = IscnInput(`
+{
+	"@context": "http://schema.org/",
+	"@type": "CreativeWorks",
+	"title": "使用矩陣計算遞歸關係式",
+	"description": "An article on computing recursive function with matrix multiplication.",
+	"datePublished": "2019-04-19",
+	"version": 1,
+	"url": "https://nnkken.github.io/post/recursive-relation/",
+	"author": "https://github.com/nnkken",
+	"usageInfo": "https://creativecommons.org/licenses/by/4.0",
+	"keywords": "matrix,recursion"
+}`)
+
+	createIscnRecord1 = IscnRecord{
+		ContentFingerprints: []string{fingerprint1},
+		Stakeholders:        []IscnInput{stakeholder1, stakeholder2},
+		ContentMetadata:     contentMetadata1,
+	}
+
+	msgCreateIscnRecord1NoNonce = MsgCreateIscnRecord{
+		From:   addr1,
+		Record: createIscnRecord1,
+	}
+	msgCreateIscnRecord1Nonce0 = MsgCreateIscnRecord{
+		From:   addr1,
+		Record: createIscnRecord1,
+		Nonce:  0,
+	}
```

Comment:
`msgCreateIscnRecord1NoNonce` and `msgCreateIscnRecord1Nonce0` are actually the same.
Not specifying field value in struct constructions implies that the field will have default value of that type, and in the case of the `Nonce` field it will be `0`.